Source/ACE/Database/CharacterDatabase.cs

```diff
@@ -117,6 +123,90 @@ public uint GetMaxId()
             return characters;
         }
 
+        public uint TokenizeByName(string characterName, AccessLevel accessLevel)
+        {
+            var result = SelectPreparedStatementAsync(CharacterPreparedStatement.CharacterSelectByName, characterName);
+            Debug.Assert(result != null);
+
+            uint boolId;
+            switch (accessLevel)
+            {
+                case AccessLevel.Advocate:
+                    boolId = (uint)PropertyBool.IsAdvocate;
+                    break;
+                case AccessLevel.Sentinel:
+                    boolId = (uint)PropertyBool.IsSentinel;
+                    break;
+                case AccessLevel.Envoy:
+                    boolId = (uint)PropertyBool.IsSentinel;
+                    break;
+                case AccessLevel.Developer:
+                    boolId = (uint)PropertyBool.IsArch;
+                    break;
+                case AccessLevel.Admin:
+                    boolId = (uint)PropertyBool.IsAdmin;
+                    break;
+                default:
+                    boolId = 0;
+                    break;
+            }
+
+            try
+            {
+                uint lowGuid = result.Result.Read<uint>(0, "guid");
+
+                if (boolId > 0)
+                    ExecutePreparedStatement(CharacterPreparedStatement.CharacterPropertiesBoolInsert, lowGuid, boolId, 1);
+                else
+                {
+                    ExecutePreparedStatement(CharacterPreparedStatement.CharacterPropertiesBoolInsert, lowGuid, (uint)PropertyBool.IsAdvocate, 0);
+                    ExecutePreparedStatement(CharacterPreparedStatement.CharacterPropertiesBoolInsert, lowGuid, (uint)PropertyBool.IsSentinel, 0);
+                    //ExecutePreparedStatement(CharacterPreparedStatement.CharacterPropertiesBoolInsert, lowGuid, (uint)PropertyBool.IsEnvoy, 0);
```

Comment:
I had forgot at the time what IsEnvoy really was, which was IsPSR.

In anycase, this PR is no good anyway. Most of it has already been rolled into master via another PR and is mostly here just to remind me to re-do character rename in another PR soon.